api/catalog_test.go

```diff
@@ -14,19 +16,17 @@ func TestCatalog_Datacenters(t *testing.T) {
 
 	catalog := c.Catalog()
 
-	if err := testutil.WaitForResult(func() (bool, error) {
+	for r := retry.R(); r.Next(t); {
 		datacenters, err := catalog.Datacenters()
 		if err != nil {
-			return false, err
+			t.Log("catalog.Datacenters: ", err)
+			continue
 		}
-
 		if len(datacenters) == 0 {
-			return false, fmt.Errorf("Bad: %v", datacenters)
+			t.Logf("got 0 datacenters want at least one")
```

Comment:
This could be a `t.Log`, though I'd always just use `t.Logf` for everything so it's less to think about when reading these, and easy to add other things to for debugging. We don't care about any performance differences at this level.